src/index.js

```diff
@@ -40,33 +57,65 @@ module.exports = class Client {
     this[PRIVATE.OFFSETS] = new Map()
     this[PRIVATE.LOGGER] = createLogger({ level: logLevel, logCreator })
     this[PRIVATE.CLUSTER_RETRY] = retry
-    this[PRIVATE.CREATE_CLUSTER] = ({
-      metadataMaxAge,
+    this[PRIVATE.CREATE_BROKERPOOL] = ({
+      metadataMaxAge = DEFAULT_METADATA_MAX_AGE,
       allowAutoTopicCreation = true,
       maxInFlightRequests = null,
       instrumentationEmitter = null,
-      isolationLevel,
-    }) =>
-      new Cluster({
+    }) => {
+      const connectionBuilder = createConnectionBuilder({
         logger: this[PRIVATE.LOGGER],
         retry: this[PRIVATE.CLUSTER_RETRY],
-        offsets: this[PRIVATE.OFFSETS],
+        instrumentationEmitter,
         socketFactory,
         brokers,
         ssl,
         sasl,
         clientId,
         connectionTimeout,
-        authenticationTimeout,
-        reauthenticationThreshold,
         requestTimeout,
         enforceRequestTimeout,
-        metadataMaxAge,
-        instrumentationEmitter,
-        allowAutoTopicCreation,
         maxInFlightRequests,
+      })
+      return new BrokerPool({
+        logger: this[PRIVATE.LOGGER],
+        retry: this[PRIVATE.CLUSTER_RETRY],
+        connectionBuilder,
+        allowAutoTopicCreation,
+        authenticationTimeout,
+        reauthenticationThreshold,
+        metadataMaxAge,
+      })
+    }
+    this[PRIVATE.CREATE_CLUSTER] = ({
+      isolationLevel,
+      brokerPool,
+      instrumentationEmitter,
+      ...brokerPoolOptions
+    }) => {
+      if (
+        Object.entries(brokerPoolOptions).filter(([, value]) => typeof value !== 'undefined')
+          .length > 0 &&
+        brokerPool
+      ) {
+        // XXX: We could compare against the actual options of the provided pool ...
+        throw new Error('Incompatible options: brokerPool and broker pool creation options')
```

Comment:
This should be a `KafkaJSNonRetriableError` and probably include some more information to understand how to resolve it.

Reply:
https://github.com/tulios/kafkajs/pull/977/commits/71883cbe5638b86ecf859238df30ed9e0e517dbe